src/emucore/Settings.cxx

```diff
@@ -672,3 +688,59 @@ Settings& Settings::operator = (const Settings&)
 
   return *this;
 }
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void Settings::setDefaultSettings() {
+
+    // Stella settings
+    stringSettings.insert(pair<string, string>("cpu", "low")); // Reduce CPU emulation fidelity for speed
+
+    // Controller settings
+    intSettings.insert(pair<string, int>("max_num_frames", 0));
+    intSettings.insert(pair<string, int>("max_num_frames_per_episode", 0));
+
+    // FIFO controller settings
+    boolSettings.insert(pair<string, bool>("run_length_encoding", true));
+
+    // Environment customization settings
+    boolSettings.insert(pair<string, bool>("restricted_action_set", false));
+    intSettings.insert(pair<string, int>("random_seed", -1));
+    boolSettings.insert(pair<string, bool>("color_averaging", true));
+    boolSettings.insert(pair<string, bool>("send_rgb", false));
+    intSettings.insert(pair<string, int>("frame_skip", 1));
+    floatSettings.insert(pair<string, float>("repeat_action_probability", 0.25));
+    stringSettings.insert(pair<string, string>("rom_file", ""));
+
+    // Record settings
+    intSettings.insert(pair<string, int>("fragsize", 64)); // fragsize to 64 ensures proper sound sync
+    stringSettings.insert(pair<string, string>("record_screen_dir", ""));
+    stringSettings.insert(pair<string, string>("record_sound_filename", ""));
+
+    // Display Settings
+    boolSettings.insert(pair<string, bool>("display_screen", false));
+
+    for(map<string, string>::iterator it = stringSettings.begin(); it != stringSettings.end(); it++) {
+      this->setString(it->first, it->second);
+    }
+
+    for(map<string, float>::iterator it = floatSettings.begin(); it != floatSettings.end(); it++) {
+      this->setFloat(it->first, it->second);
+    }
+
+    for(map<string, bool>::iterator it = boolSettings.begin(); it != boolSettings.end(); it++) {
+      this->setBool(it->first, it->second);
+    }
+
+    for(map<string, int>::iterator it = intSettings.begin(); it != intSettings.end(); it++) {
+      this->setInt(it->first, it->second);
+    }
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+template<typename TYPE>
+void Settings::verifyVariableExistence(map<string, TYPE> dict, string key){
+    if(dict.find(key) == dict.end()){
+      cerr << "The key " << key << " you are trying to set does not exist." << endl;
```

Comment:
Sure. Since we never agreed on what to do with errors, I used the same pattern that is in this file when someone tries to get a variable that does not exist.